packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreDuplicateRoutes.spec.ts

```diff
@@ -0,0 +1,154 @@
+            newSummaryContext.proposalHandle = latestAckedSummary.summaryOp.contents.handle;
+        }
+        return newSummaryContext;
+    }
+
+    const logger = new TelemetryNullLogger();
+
+    async function summarizeOnNewContainerAndGetGCTree(summaryVersion?: string): Promise<IGarbageCollectionState> {
+        await provider.ensureSynchronized();
+        const summarizerClient = await loadSummarizer(
+            provider,
+            runtimeFactory,
+            mainContainer.deltaManager.lastSequenceNumber,
+            summaryVersion,
+            { configProvider },
+        );
+        const summaryResult = await submitAndAckSummary(provider, summarizerClient, logger, false /* fullTree */);
+        latestAckedSummary = summaryResult.ackedSummary;
+        assert(latestUploadedSummary !== undefined, "Did not get a summary");
+        return getGCContentFromTree(latestUploadedSummary);
+    }
+
+    const createContainer = async (): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory, { configProvider });
+    };
+
+    const getGCContentFromTree = (summary: ISummaryTree): IGarbageCollectionState => {
+        const gcTree: ISummaryTree = summary.tree[gcTreeKey] as ISummaryTree;
+        const gcBlob = gcTree.tree[`${gcBlobPrefix}_root`] as ISummaryBlob;
+        const content = JSON.parse(gcBlob.content as string) as IGarbageCollectionState;
+        return content;
+    };
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider();
+        // Wrap the document service factory in the driver so that the `uploadSummaryCb` function is called every
+        // time the summarizer client uploads a summary.
+        (provider as any)._documentServiceFactory = wrapDocumentServiceFactory(
+            provider.documentServiceFactory,
+            uploadSummaryCb,
+        );
+
+        // Create a Container for the first client.
+        mainContainer = await createContainer();
+        mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await provider.ensureSynchronized();
+    });
+
+    it("Back routes added by GC are removed when passed from data stores to DDSs", async () => {
+        const dds = SharedMap.create(mainDataStore.dataStoreRuntime);
+        mainDataStore._root.set("dds", dds.handle);
+
+        await summarizeOnNewContainerAndGetGCTree();
+
+        // Change ds1 but not the root dds
+        dds.set("change", "change1");
+
+        assert(latestAckedSummary !== undefined, "Ack'd summary isn't available as expected");
+        const gcTree = await summarizeOnNewContainerAndGetGCTree(latestAckedSummary.summaryAck.contents.handle);
+        assert(gcTree !== undefined, "Expected a gc tree!");
+        const routes = gcTree.gcNodes[`/${mainDataStore.id}/${mainDataStore._root.id}`].outboundRoutes;
+        const seenRoutes = new Set<string>();
+        routes.forEach((route) => {
```

Comment:
It may be worth checking that none of the nodes have duplicate routes. Maybe a helper function that does this. Eventually we can make that part of all our GC tests that validate the GC data. That will give us coverage in more scenarios.